Assets/Scripts/LoadingScreen.cs

```diff
@@ -0,0 +1,69 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class LoadingScreen : MonoBehaviour
+{
+    public float loadTime = 10;
+    public Slider ProgressBar;
+    public List<LoadedAsset> Assets;
+    public List<LoadedAsset> ResourcesAssets;
+    public List<LoadedAsset> StreamingAssets;
```

Comment:
try to change you solution to be more object oriented, have a class that will implement some "Load()" function
and have two derived classed for the different ways of loading.

this will allow you to have just one list of "assets"